hathor/indexes/tips_index.py

```diff
@@ -38,6 +56,20 @@ class TipsIndex(BaseIndex):
     TODO Use an interval tree stored in disk, possibly using a B-tree.
     """
 
+    def __init__(self, *, txs: bool = False, blocks: bool = False, all: bool = False):
+        if sum([txs, blocks, all]) != 1:
+            raise TypeError('Exactly one of these: txs, blocks, all, must be set to True')
+        self._scope: Scope
+        if txs:
+            self._scope = SCOPE_TXS
+        elif blocks:
+            self._scope = SCOPE_BLOCKS
+        elif all:
+            self._scope = SCOPE_ALL
```

Comment:
It looks like you could benefit from using an Enum here, since only one option can be chosen:

```python
    class ScopeType(Enum):
        ALL = Scope(
            include_blocks=True,
            include_txs=True,
            include_voided=True,
        )
        
        TXS = Scope(
            include_blocks=False,
            include_txs=True,
            include_voided=False,
        )
        
        BLOCKS = Scope(
            include_blocks=True,
            include_txs=False,
            include_voided=True,
        )
```

And then:

```python
    def __init__(self, *, scope: ScopeType):
        self._scope: Scope = scope.value
```

That would also simplify some subclasses, since `name` just becomes `name = scope.name.lower()`

Reply:
Oh, that's a good point. It adds a symbol to be imported (`ScopeType`), which I'll make a little longer because I don't want to tie the specific values between different indexes (timestamp and tips indexes only incidentally have the same values, we might refactor one of them in a way that the scope changes). So I think I'll make it `TipsScopeType` and `TimestampScopeType`, or at least import them as that.

Reply:
Sounds good!

Reply:
Done.